### PR TITLE
Upgrade to React 18

### PR DESCRIPTION
**Pull request by michael-markl**
To complete the upgrade to react 18, we need to use createRoot from react-dom/client.
Moreover, I had to adjust the creation of the AppToaster, which uses the deprecated old API.

### CODE REVIEW

File: administration/src/components/AppToaster.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+/** Singleton toaster instance. Create separate instances for different options. */
+let toasterRef: Toaster | null
+createRoot(document.getElementById("toaster")!).render(
+    <Toaster ref={ref => {
+        toasterRef = ref
+    }} position={Position.TOP}/>
+)
+export const getAppToaster = () => {
+    return toasterRef!
+}
```
Comment (maxammann) [inline administration/src/components/AppToaster.tsx:13]: Oke this is weird. so the toaster has an own custom virtual dom? weird.

**Review by maxammann — APPROVED** [review]

File: administration/src/components/AppToaster.tsx (hunk 1 above)
Comment (michael-markl) [reply administration/src/components/AppToaster.tsx]: I agree! I adjusted it to use the same DOM now. Maybe you can take another look.

**Review by maxammann — APPROVED** [review]
I think this is better, if no null ptr exceptions can happen 👍🏼

**Comment by michael-markl** [comment]
Yeah, there shouldn't be any (TM).

**Pull request merged**